kola/tests/locksmith/locksmith.go

```diff
@@ -117,6 +121,27 @@ func init() {
 	})
 }
 
+func locksmithReboot(c cluster.TestCluster) {
+	// The machine should be able to reboot without etcd in the default mode
+	m := c.Machines()[0]
+
+	output, err := m.SSH("locksmithctl send-need-reboot")
+	if _, ok := err.(*ssh.ExitMissingError); ok {
+		err = nil // A terminated session is perfectly normal during reboot.
+	} else if err == io.EOF {
+		err = nil // Sometimes copying command output returns EOF here.
+	}
+	if err != nil {
+		c.Fatalf("failed to run \"locksmothctl send-need-reboot\": output: %q status: %q", output, err)
+	}
+
+	err = platform.CheckMachine(m)
```

Comment:
This doesn't confirm that the machine has rebooted yet, or in fact will reboot at all.  `/proc/sys/kernel/random/boot_id` may help here.